bindings/py/tests/sparse_matrix_test.py

```diff
@@ -162,6 +162,70 @@ def test_setAllNonZeros(self):
         error('setAllNonZeros, out of order')
 
 
+  def test_nNonZerosPerRow_allRows(self):
+    for name, denseMatrix, expected in (
+        ("Test 1",
+         numpy.array([[1, 1, 0, 0, 1, 1],
+                      [0, 0, 1, 1, 0, 0],
+                      [0, 0, 1, 0, 0, 1],
+                      [1, 0, 1, 1, 0, 0],
+                      [0, 0, 0, 0, 0, 1],
+                      [0, 0, 0, 0, 0, 0],
+                      [1, 1, 1, 1, 1, 1],
+                      [0, 0, 1, 1, 0, 1]], dtype="float32"),
+         [4, 2, 2, 3, 1, 0, 6, 3]),
+        ("No rows",
+         numpy.zeros((0,0), dtype="float32"),
+         []),
+        ("Empty rows",
+         numpy.array([[],
+                      [],
+                      []], dtype="float32"),
+         [0, 0, 0]),
+    ):
+      m = SparseMatrix(denseMatrix)
+
+      numpy.testing.assert_equal(m.nNonZerosPerRow(),
```

Comment:
Yeah, and it produces good error messages. For example:

~~~
E       AssertionError:
E       Arrays are not equal
E
E       (mismatch 25.0%)
E        x: array([4, 0, 6, 3], dtype=uint32)
E        y: array([3, 0, 6, 3])
~~~

If there's more to being unittest-aware, let me know. I'm assuming it just means "raise AssertionErrors with useful messages".